### ISSUE TITLE
Uninformative error when using external function in domain axiom

### ISSUE THREAD

**Issue by fpoli** [fpoli/viper-silver#204]
When using a function inside a domain axiom, which is not declared inside the domain and uses the domain type (e.g. like in the example below), the Viper frontend throws the following error, which is very uninformative:
`java.lang.IllegalArgumentException: requirement failed`

```
#!silver

domain Test {
    axiom bar {
        forall t: Test :: foo(t)
    }
}

function foo(x: Test): Bool
```

[Issue created by tierriminator: 2018-07-18]
[Last updated on bitbucket: 2018-07-30]

[Comment created by alexander_summers: 2018-07-30]
Duplicate of #238.

**Comment by fpoli** [comment]
> **@vakaras** commented on 2017-05-31 08:59

Desugared code:

```
#!silver
field val: Int

method test1(head: Ref)
{
  package (forall n: Ref :: acc(n.val, write)) --* true
  exhale (forall n$0: Ref :: acc(n$0.val, write)) --* true
}

method test2(head: Ref)
{
  package (forall n: Ref :: acc(n.val, write)) --* true
  exhale (forall n: Ref :: acc(n.val, write)) --* true
}

```

**Comment by fpoli** [comment]
> **@mschwerhoff** commented on 2017-06-18 11:19

**Comment by fpoli** [comment]
> **@mschwerhoff** commented on 2017-06-18 11:20

Consistency checks should reject this combination of features until it is officially supported

**Comment by fpoli** [comment]
> **@mschwerhoff** commented on 2017-11-22 15:02

Resolved with commit 067548eef290
